framework/auth/core.py

```diff
@@ -481,14 +502,19 @@ def create(cls, username, password, fullname):
         return user
 
     @classmethod
-    def create_unconfirmed(cls, username, password, fullname, do_confirm=True,
+    def create_unconfirmed(cls, username, password, fullname, oauth=None, do_confirm=True,
                            campaign=None):
         """Create a new user who has begun registration but needs to verify
         their primary email address (username).
         """
         user = cls.create(username, password, fullname)
         user.add_unconfirmed_email(username)
         user.is_registered = False
+        if oauth:
+            if user.oauth:
```

Comment:
A migration shouldn't be necessary -- if it's not present in the database document, the default value will be assumed. 
On `develop`:
```
In [1]: User.find()[0].oauth
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-798447f85e8d> in <module>()
----> 1 User.find()[0].oauth

/Users/matt/.virtualenvs/osf/lib/python2.7/site-packages/modularodm/storedobject.pyc in wrapped(this, *args, **kwargs)
     93         if '_detached' in this.__dict__ and this._detached:
     94             warnings.warn('here')
---> 95         return func(this, *args, **kwargs)
     96     return wrapped
     97

/Users/matt/.virtualenvs/osf/lib/python2.7/site-packages/modularodm/storedobject.pyc in __getattr__(self, item)
    933             return ForeignList(ids, literal=True, base_class=base_class)
    934
--> 935         raise AttributeError(errmsg)
    936
    937     @warn_if_detached

AttributeError: User object has no attribute oauth
```
On this PR branch:
```
In [1]: User.find()[0].oauth
Out[1]: {}
```